packages/engine/vscode/src/lib/terminal.ts

```diff
@@ -0,0 +1,94 @@
+import { Plugin } from "@remixproject/engine";
+import { window, Terminal, OutputChannel } from 'vscode';
+
+export interface TerminalOptions {
+  name?: string
+  open: boolean
+}
+
+function getOptions(params: Partial<TerminalOptions> = {}): TerminalOptions {
+  return {
+    open: true,
+    ...params
+  }
+}
+
+export class TerminalPlugin extends Plugin {
+  // Terminals (stdin) created during activation 
+  private terminals: Record<string, Terminal> = {};
+  // Outputs (stdout) created during activation
+  private outputs: Record<string, OutputChannel> = {};
+  // Name of the active output
+  private activeOutput: string;
+
+  constructor() {
+    super({ name: 'terminal', methods: ['write', 'exec', 'open', 'kill'] })
+  }
+
+  onDeactivation() {
+    Object.values(this.terminals).forEach(terminal => terminal.dispose());
+    this.terminals = {};
+    Object.values(this.outputs).forEach(output => output.dispose());
+    this.outputs = {};
+  }
+
+  private get active() {
+    return window.activeTerminal;
+  }
+
+  private getTerminal(name?: string) {
+    if (name) {
+      const terminal = window.terminals.find(terminal => terminal.name === name);
+      if (!terminal) {
+        return this.terminals[name] = window.createTerminal(name);
+      }
+      return terminal || window.createTerminal(name);
+    }
+    return this.active;
+  }
+  
+  private getOutput(name?: string) {
+    name = name ?? this.activeOutput ?? 'plugin-engine';
+    return this.outputs[name] ?? window.createOutputChannel(name);
+  }
+
+  /** Open specific terminal (doesn't work with output) */
```

Comment:
`doesn't work with output` what does that exactly means?

Reply:
In vscode outputs (stdout) & terminal (stdin) are two different tabs. When you run `open` it can open a terminal but not an output. I could add a second parameter to specify if you want to open a terminal or output. But it's very specific to vscode and if we want to use this API in remix-ide for example, this params wouldn't make sense.
I would prefer having some feedback from plugin developers before going to deep into this implementation.